src/c++/tests/cc_client_test.cc

```diff
@@ -1613,6 +1617,491 @@ TEST_F(GRPCTraceTest, GRPCClearTraceSettings)
       << std::endl;
 }
 
+class TestHttpInferRequest : public tc::HttpInferRequest {
+ public:
+  tc::Error ConvertBinaryInputsToJSON(
+      tc::InferInput& input,
+      triton::common::TritonJson::Value& data_json) const override
+  {
+    return tc::HttpInferRequest::ConvertBinaryInputsToJSON(input, data_json);
+  }
+
+  tc::Error ConvertBinaryInputToJSON(
+      const uint8_t* buf, const size_t element_count,
+      const std::string& datatype,
+      triton::common::TritonJson::Value& data_json) const override
+  {
+    return tc::HttpInferRequest::ConvertBinaryInputToJSON(
+        buf, element_count, datatype, data_json);
+  }
+};
+
+TEST_F(HTTPJSONDataTest, ConvertBinaryInputsToJSON)
```

Comment:
Is there a best practice to comment on the intent of the test? Either through doxygen comments or something built in to the testing macros?

Reply:
I think this needs to be broken up a bit with comments as well. its hard to parse what you are testing and where.

Reply:
Added comments to make it clearer what's being tested 👍